datastore/mongodb/error.go

```diff
@@ -52,3 +53,9 @@ func (e *er) Unavailable() bool {
 func (e *er) Forbidden() bool {
 	return e.forbidden
 }
+
+// BadRequest satisfies the badRequest interface and allows other packages
```

Comment:
Mongo shouldn't know about http requests. If we can't find the dimension with the name we should return a notFound which should then translate to a 404

Reply:
the rest of the error types in that file are essentially in HTTP request language, so i followed the convention there. 
The swagger spec also defines a missing dimension name as a 400, and a 404 only when filter not found

Reply:
as discussed, I have removed the additions to the interfaces, and just worked with what is there. New solution pushed. I will let you resolve if you are happy